Source/WebCore/platform/network/curl/NetworkStorageSessionCurl.cpp

```diff
@@ -77,11 +84,15 @@ static std::pair<String, bool> cookiesForSession(const NetworkStorageSession& se
     return { cookies.toString(), didAccessSecureCookies };
 }
 
-NetworkStorageSession::NetworkStorageSession(PAL::SessionID sessionID)
+NetworkStorageSession::NetworkStorageSession(PAL::SessionID sessionID, const String& alternativeServicesDirectory)
     : m_sessionID(sessionID)
     // :memory: creates in-memory database, see https://www.sqlite.org/inmemorydb.html
     , m_cookieDatabase(makeUniqueRef<CookieJarDB>(sessionID.isEphemeral() ? ":memory:"_s : defaultCookieJarPath()))
 {
+    if (!alternativeServicesDirectory.isEmpty()) {
+        FileSystem::makeAllDirectories(alternativeServicesDirectory);
```

Comment:
I think you'll find this is unnecessary because it's done in the UI process before telling the network process to make this NetworkStorageSession.  It's in cURL-specific code, though, so I don't have strong opinions.

Reply:
I tried it, but the directory was not created. Since SandboxExtension is not enabled outside of Cocoa, it seems that the directory is not created. So, I leave this code as-is.